src/main/flight/imu.c

```diff
@@ -609,6 +606,25 @@ static void imuCalculateEstimatedAttitude(timeUs_t currentTimeUs)
 }
 #else
 
+#if defined(USE_GPS)
+static void isGpsHeadingUsable(float groundspeedGain, float imuCourseError, float dt)
+{
+    // ** ATTEMPT TO DETECT SUCCESSFUL IMU ORIENTATION TO GPS **
```

Comment:
This is far too magic.
It should be quite easy to use Kalman (or Kalman-like) estimator for heading, IIRC there is open PR for it

Reply:
Sure if someone can improve our current method, and have it report when the IMU is likely to have a valid heading from GPS, that would be fantastic.  

However we have existing code, and it works OK, and with this PR I get a lot of pressure to allow position hold without a Mag.  Hence we have a need, right now, for a method to identify when the IMU is likely to have a valid heading from GPS alone.

Whether we do this with 'magic' or a Kalman that returns likely precision of the estimate, I don't care.  I only propose a solution like this on the basis that it is possible for me to code this, but I cannot code a Kalman.

Reply:
The problem is that it takes huge effort to remove these magic parts. There is no physical background, no math, just equations hacked until slightly sensible results appear. And these hacks do spread through code, eventually making it unmaintainable.

Reply:
PS this is not yet tested at the field so I don't know for sure that it works; let's not worry too much about it for now.

But the danger is that if a user has no Mag, position hold code will rapidly fly away in an unpredictable direction if the IMU heading is significantly wrong.  So a user may take off, hit position hold, and have an immediate rapid flyaway without having something to protect them.  That's why I am trying to make some solution to solve this, otherwise for sure lots of people will have this problem.

Reply:
There is already position check. It may be enough until there is clean way to detect usable heading information.